src/components/OptionRow.js

```diff
@@ -75,167 +76,191 @@ const defaultProps = {
-                                        src={Expensicons.DotIndicator}
-                                        fill={themeColors.danger}
-                                        height={variables.iconSizeSmall}
-                                        width={variables.iconSizeSmall}
-                                    />
+                                    )}
+                                    {this.props.showSelectedState && <SelectCircle isChecked={this.props.isSelected} />}
                                 </View>
-                                )}
-                                {props.showSelectedState && <SelectCircle isChecked={props.isSelected} />}
                             </View>
-                        </View>
-                        {props.option.customIcon && (
-                            <View
-                                style={[styles.flexRow, styles.alignItemsCenter]}
-                                accessible={false}
-                            >
-                                <View>
-                                    <Icon
-                                        src={lodashGet(props.option, 'customIcon.src', '')}
-                                        height={16}
-                                        width={16}
-                                        fill={lodashGet(props.option, 'customIcon.color')}
-                                    />
+                            {this.props.option.customIcon && (
+                                <View
+                                    style={[styles.flexRow, styles.alignItemsCenter]}
+                                    accessible={false}
+                                >
+                                    <View>
+                                        <Icon
+                                            src={lodashGet(this.props.option, 'customIcon.src', '')}
+                                            height={16}
+                                            width={16}
+                                            fill={lodashGet(this.props.option, 'customIcon.color')}
+                                        />
+                                    </View>
                                 </View>
-                            </View>
-                        )}
-                    </TouchableOpacity>
-                )}
-            </Hoverable>
-        </OfflineWithFeedback>
-    );
-};
+                            )}
+                        </TouchableOpacity>
+                    )}
+                </Hoverable>
+            </OfflineWithFeedback>
+        );
+    }
+}
 
 OptionRow.propTypes = propTypes;
 OptionRow.defaultProps = defaultProps;
-OptionRow.displayName = 'OptionRow';
 
 // It it very important to use React.memo here so SectionList items will not unnecessarily re-render
 export default withLocalize(memo(OptionRow, (prevProps, nextProps) => prevProps.optionIsFocused === nextProps.optionIsFocused
```

Comment:
Since this is a class-based component, we can avoid unnecessarily re-renders by using `shouldComponentUpdate`

Reply:
```suggestion
export default withLocalize(OptionRow);
```

> Since this is a class-based component, we can avoid unnecessarily re-renders by using `shouldComponentUpdate`
We can write shouldComponentUpdate like this.

```diff --git a/src/components/OptionRow.js b/src/components/OptionRow.js
index 3be19f08a..7dad7f589 100644
--- a/src/components/OptionRow.js
+++ b/src/components/OptionRow.js
@@ -92,6 +92,19 @@ class OptionRow extends Component {
         this.setState({isDisabled: this.props.isDisabled});
     }
 
+    shouldComponentUpdate(prevProps, nextProps) {
+        return prevProps.optionIsFocused === nextProps.optionIsFocused
+            && prevProps.isSelected === nextProps.isSelected
+            && prevProps.option.alternateText === nextProps.option.alternateText
+            && prevProps.option.descriptiveText === nextProps.option.descriptiveText
+            && _.isEqual(prevProps.option.icons, nextProps.option.icons)
+            && prevProps.option.text === nextProps.option.text
+            && prevProps.showSelectedState === nextProps.showSelectedState
+            && prevProps.isDisabled === nextProps.isDisabled
+            && prevProps.showTitleTooltip === nextProps.showTitleTooltip
+            && prevProps.option.brickRoadIndicator === nextProps.option.brickRoadIndicator
+    }    
+
     render() {
         let touchableRef = null;
         const textStyle = this.props.optionIsFocused
```

Reply:
Done

Reply:
@eVoloshchak Is there a specific reason that we didn't go for a `PureComponenet` here?

Reply:
@s77rt As pureCompnent performs a shallow comparison, it may not work as expected for complex data structures like objects and arrays. In these cases, you need to implement your own shouldComponentUpdate method to perform a deep comparison. I think that was the reason we didn't opted for pureComponent.

Reply:
Right, but shallow copy is not necessary bad thing here, is it? The worst thing can happen is an unnecessary render which is better than the worst thing being a missing render.
How was the render process using functional component? And did you face any issue with PureComponent?

Reply:
On the other hand I see your point here. Using `shouldComponentUpdate` would be best as long as we cover all the cases that should be covered. I think we will revisit the props and add missing conditions. Thanks for your quick response here!